### PR TITLE
Use VLQ and ZigZag encoding for putShort/getShort

### PR DESCRIPTION
**Pull request by aslesarenko**
Also some code style fixes.
The following checked to work with this new version:
- ergo-wallet
- sigma-state
- ergo unit tests

### CODE REVIEW

File: src/main/scala/scorex/util/serialization/VLQWriter.scala [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
+    * Use [[putUShort]] to encode values that are always positive.
+    *
+    * @see [[https://en.wikipedia.org/wiki/Variable-length_quantity]]
+    * @note Have to be decoded '''only''' with [[VLQReader.getShort]]
+    *       The resulting varint uses ZigZag encoding, which is much more efficient at
+    *       encoding negative values than pure VLQ.
+    * @param x signed Short
+    */
+  @inline override def putShort(x: Short): this.type = {
+    if (x < Short.MinValue || x > Short.MaxValue)
```
Comment (kushti) [inline src/main/scala/scorex/util/serialization/VLQWriter.scala]: This check is not needed, as the value in the Short domain.

File: src/main/scala/scorex/util/serialization/VLQWriter.scala [hunk 2]
```diff
@@ -1,23 +1,43 @@
...
+      throw new ArithmeticException(s"Cannot convert Int to Short due to overflow: $x")
+    putULong(encodeZigZagInt(x.toShort))
```
Comment (kushti) [inline src/main/scala/scorex/util/serialization/VLQWriter.scala]: .toShort is not needed

**Review by kushti — CHANGES_REQUESTED** [review]

**Comment by kushti** [comment]
@aslesarenko also, please avoid assert as it is elidable. Use require, or throw an exception

File: src/main/scala/scorex/util/serialization/VLQWriter.scala (hunk 1 above)
Comment (aslesarenko) [reply src/main/scala/scorex/util/serialization/VLQWriter.scala]: fixed

File: src/main/scala/scorex/util/serialization/VLQWriter.scala (hunk 2 above)
Comment (aslesarenko) [reply src/main/scala/scorex/util/serialization/VLQWriter.scala]: fixed

**Comment by aslesarenko** [comment]
All issues are fixed.

**Review by kushti — APPROVED** [review]

**Pull request merged**